src/rest.cpp

```diff
@@ -792,13 +795,14 @@ static bool rest_getutxos(const std::any& context, HTTPRequest* req, const std::
             if (txid_out.size() != 2) {
                 return RESTERR(req, HTTP_BAD_REQUEST, "Parse error");
             }
+            auto txid{Txid::FromHex(txid_out.at(0))};
             auto output{ToIntegral<uint32_t>(txid_out.at(1))};
 
-            if (!output || !IsHex(txid_out.at(0))) {
+            if (!txid || !output) {
```

Comment:
Since this is a potentially breaking interface change for downstream applications, adding release notes might be appropriate?

<details>
<summary>git diff on fa2d94a01c</summary>

```diff
diff --git a/doc/release-notes-30482.md b/doc/release-notes-30482.md
new file mode 100644
index 0000000000..6eb8af3f69
--- /dev/null
+++ b/doc/release-notes-30482.md
@@ -0,0 +1,6 @@
+Updated REST APIs
+-----------------
+- Parameter validation for `/rest/getutxos` has been improved by
+  rejecting truncated or overly large txids by raising an
+  HTTP_BAD_REQUEST "Parse error". Previously, these malformed txids
+  would be silently ignored.

```
</details>